### PR TITLE
ARTEMIS-1548 Support CRL

### PR DESCRIPTION
**Pull request by Skiler**

### CODE REVIEW

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt/imported/MQTTSecurityCRLTest.java [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+
+   private BlockingConnection retrieveMQTTConnection(String host, String truststorePath, String truststorePass, String keystorePath, String keystorePass) throws Exception {
+
+      MQTT mqtt = new MQTT();
+      mqtt.setConnectAttemptsMax(1);
+      mqtt.setReconnectAttemptsMax(0);
+      mqtt.setHost(host);
+      mqtt.setUserName(fullUser);
+      mqtt.setPassword(fullPass);
+      SSLContext sslContext = SSLContext.getInstance("TLS");
```
Comment (jbertram) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt/imported/MQTTSecurityCRLTest.java]: Why not use the SSLSupport class that's already included in the code-base to generate the SSLContext?

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt/imported/MQTTSecurityCRLTest.java [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+
+      // we need this otherwise the data folder will be located under activemq-server and not on the temporary directory
+      fc.setPagingDirectory(getTestDir() + "/" + name + "/" + fc.getPagingDirectory());
+      fc.setLargeMessagesDirectory(getTestDir() + "/" + name + "/" + fc.getLargeMessagesDirectory());
+      fc.setJournalDirectory(getTestDir() + "/" + name + "/" + fc.getJournalDirectory());
+      fc.setBindingsDirectory(getTestDir() + "/" + name + "/" + fc.getBindingsDirectory());
+
+      addMqttTransportConfiguration(fc);
+      addWildCardConfiguration(fc);
+      fc.setSecurityEnabled(true);
```
Comment (jbertram) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt/imported/MQTTSecurityCRLTest.java]: Is enabling security really necessary?  Isn't this test simply exercising the CRL functionality which happens at the transport layer (i.e. SSL handshake) and not at the broker's security layer?

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt/imported/MQTTSecurityCRLTest.java [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+
+   private ActiveMQServer initServer() throws Exception {
+      Configuration configuration = createConfiguration("broker");
+      ActiveMQServer server = createServer(true, configuration);
+      configureBrokerSecurity(server);
+      return server;
+   }
+
+   protected Configuration createConfiguration(String name) throws Exception {
+      FileConfiguration fc = new FileConfiguration();
```
Comment (jbertram) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/mqtt/imported/MQTTSecurityCRLTest.java]: There are already lots of methods in the test-suite to create a config for a test.  Take a look at org.apache.activemq.artemis.tests.util.ActiveMQTestBase#createDefaultNettyConfig for example.

**Comment by jbertram** [comment]
Can you squash the commits?

**Comment by Skiler** [comment]
Done :)

**Comment by Skiler** [comment]
Hi @jbertram 

Do I need to do anything else?

Thanks in advance

**Comment by jbertram** [comment]
Rather than go back and forth on this again and again I just modified your commit and then squashed my modifications before I merged.  Modifications include:

* Changing example docs to markdown format from HTML
* Simplifying example to use SSLSupport rather than manually creating the SSLContext
* Simplifying example to not use authorization since the example was only for SSL related security
* Simplifying test to use keystores on the classpath rather than manual paths
* Simplifying test to not use authorization since the test was only for SSL related security

There were a handful of other things in there...

**Comment by Skiler** [comment]
Ok. Thanks :)

**Pull request closed without merging**